src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.ChangeNamespaceCodeAction.cs

```diff
@@ -0,0 +1,526 @@
+                        .SelectMany(refSymbol => refSymbol.Locations);
+
+                    foreach (var refLocation in refLocationsForSymbol)
+                    {
+                        if (refLocation.Document.Id == document.Id)
+                        {
+                            refLocationsInCurrentDocument.Add(refLocation);
+                        }
+                        else
+                        {
+                            Debug.Assert(!PathUtilities.PathsEqual(refLocation.Document.FilePath, document.FilePath));
+                            refLocationsInOtherDocuments.Add(refLocation);
+                        }
+                    }
+                }
+
+                document = await FixDeclarationDocumentAsync(document, refLocationsInCurrentDocument, cancellationToken)
+                    .ConfigureAwait(false);
+                solution = document.Project.Solution;
+
+                var refLocationGroups = refLocationsInOtherDocuments.GroupBy(loc => loc.Document.Id);
+                foreach (var refInOneDocument in refLocationGroups)
+                {
+                    var refDocument = solution.GetDocument(refInOneDocument.Key);
+                    document = await FixReferencingDocumentAsync(refDocument, refInOneDocument, cancellationToken)
+                        .ConfigureAwait(false);
+                    solution = document.Project.Solution;
+                }
+
+                return (solution, refLocationGroups.SelectAsArray(g => g.Key));
+            }            
+
+            private async Task<Document> FixDeclarationDocumentAsync(
+                Document document, 
+                IReadOnlyList<ReferenceLocation> refLocations, 
+                CancellationToken cancellationToken)
+            {
+                Debug.Assert(!NewNamespaceParts.IsDefault);
+
+                // 1. Fix references to the affected types in this document if necessary.
+                // 2. Add usings for containing namespaces, in case we have references 
+                //    relying on old namespace declaration for resolution. 
+                //
+                //      For example, in the code below, after we change namespace to 
+                //      "A.B.C", we will need to add "using Foo.Bar;".     
+                //
+                //      namespace Foo.Bar.Baz
+                //      {
+                //          class C1
+                //          {
+                //               C2 _c2;    // C2 is define in namespace "Foo.Bar" in another document.
+                //          }
+                //      }
+                //
+                // 3. Change namespace declaration to target namespace.
+                // 4. Simplify away unnecessary qualifications.
+
+                var addImportService = document.GetLanguageService<IAddImportsService>();
+                ImmutableArray<SyntaxNode> containers;
+                SyntaxNode root;
```

Comment:
root should not be declared here.  you clearly don't set it through all these codepaths, so it couldn't be used as the result of setting it in the 'else' block.